rename: Connectors -> LiquidityPools

NunoAlexandre opened the pull request:
Closes https://github.com/centrifuge/centrifuge-chain/issues/1489

### Notes

I tried to rename "connectors" to "liquidityPools" 1:1 as much as possible but in certain cases that didn't make a lot of sense. For example, when adding a "known connector", I rephrased that as "instance", since "known liquidity pool" is not quite right in my pov. Happy to be told otherwise.

offerijns reviewed (approved):
LGTM. I think the instance terminology makes sense!

wischli commented on Cargo.toml:
```diff
@@ -38,9 +38,9 @@ members = [
-  "pallets/connectors",
-  "pallets/connectors-gateway",
-  "pallets/connectors-gateway/routers",
+  "pallets/liquidity-pools",
+  "pallets/liquidity-pools-gateway",
+  "pallets/liquidity-pools-gateway/routers",
```
Could you please apply lexicographical ordering? Thanks!

NunoAlexandre replied:
I am not sure how you would like this order to be different? Or do you mean your comment for the whole file?

wischli replied:
I was referring to moving down `pallets/liquidity-pools-*` some lines down as the next pallet is `claims` which is lexicographically smaller.

NunoAlexandre replied:
Done

wischli commented on libs/types/src/domain_address.rs:
```diff
@@ -19,15 +19,15 @@ use sp_std::{vec, vec::Vec};
 pub enum Domain {
 	/// Referring to the Centrifuge Parachain. Will be used for handling
-	/// incoming messages. NOTE: Connectors messages CAN NOT be sent directly
+	/// incoming messages. NOTE: messages CAN NOT be sent directly
```
Ultra nit: 
```suggestion
	/// incoming messages. 
	///
	/// NOTE: Messages CAN NOT be sent directly
```

wischli commented on pallets/liquidity-pools-gateway/routers/Cargo.toml:
```diff
@@ -65,7 +65,7 @@ runtime-benchmarks = [
   "frame-support/runtime-benchmarks",
   "frame-system/runtime-benchmarks",
-  "pallet-connectors-gateway/runtime-benchmarks",
+  "pallet-liquidity-pools-gateway/runtime-benchmarks",
```
Could you please apply lexicographical ordering? Thanks!

wischli commented on pallets/liquidity-pools-gateway/src/lib.rs:
```diff
@@ -139,17 +139,14 @@ pub mod pallet {
-		/// Maximum number of connectors for a domain was reached.
-		MaxConnectorsReached,
+		/// Maximum number of instances for a domain was reached.
+		MaxDomainInstances,
```
I really like this renaming!

NunoAlexandre replied:
I like you 🤟

wischli commented on pallets/liquidity-pools-gateway/src/lib.rs:
```diff
@@ -178,43 +175,44 @@ pub mod pallet {
-				ConnectorsAllowlist::<T>::contains_key(connector.domain(), connector.clone()),
-				Error::<T>::ConnectorNotFound,
+				Allowlist::<T>::contains_key(instance.domain(), instance.clone()),
+				Error::<T>::UnknownInstance,
```
Thanks for merging both errors. Totally makes sense as both were thrown when the instance was not part of the allowlist.

wischli commented on pallets/liquidity-pools/src/contract.rs:
```diff
@@ -13,11 +13,11 @@
 use sp_std::{vec, vec::Vec};
 
-/// The ConnectorsXcmContract handle function name.
+/// The solidity LiquidityPool's XCMRouter handle function name.
```
Ultra nit: 

```suggestion
/// The solidity LiquidityPools' XCMRouter handle function name.
```

wischli commented on pallets/liquidity-pools/src/message.rs:
```diff
@@ -17,16 +17,16 @@ pub const TOKEN_NAME_SIZE: usize = 128;
-/// are big-endian encoded and enum values (such as `[crate::Domain]`) also have
-/// a custom CGMPF implementation, aiming for a fixed-size encoded
+/// A message requires a custom decoding & encoding, meeting the
+/// LiquidityPool Generic Message Passing Format (GMPF): Every message is
```
So we decided on renaming CGMPF to GMPF?

NunoAlexandre replied:
I mean, it's not a name it's just an abbreviation. With "Connectors" we abbreviated it to `[C]onnectors[G]eneral[M]essage[P]assing[F]ormat` but for the sake of avodiding crazy and ad-hoc abbreviations I didn't make it `LQGMPF` but just used the general `GMPF` here. Hope that makes sense!

wischli replied:
I fully agree. The main reason for the comment was to raise awareness to other potential reviewers. I know that these huge renaming refactors sometimes make unwanted changes which can be easily overlooked.

NunoAlexandre replied:
I see :) thanks!

wischli commented on runtime/common/Cargo.toml:
```diff
@@ -74,7 +74,7 @@ std = [
   "pallet-authorship/std",
   "pallet-balances/std",
-  "pallet-connectors/std",
+  "pallet-liquidity-pools/std",
```
Could you please apply lexicographical ordering? Thanks! :)

wischli commented on runtime/development/Cargo.toml:
```diff
@@ -160,7 +160,7 @@ std = [
   "chainbridge/std",
   "codec/std",
-  "connectors-gateway-routers/std",
+  "liquidity-pools-gateway-routers/std",
```
Could you please apply lexicographical ordering? Thanks! :)

wischli commented on runtime/development/src/lib.rs:
```diff
@@ -1927,14 +1927,14 @@ construct_runtime!(
 		LiquidityRewardsBase: pallet_rewards::<Instance1>::{Pallet, Storage, Event<T>, Config<T>} = 106,
 		LiquidityRewards: pallet_liquidity_rewards::{Pallet, Call, Storage, Event<T>} = 107,
-		Connectors: pallet_connectors::{Pallet, Call, Storage, Event<T>} = 108,
+		LiquidityPools: pallet_liquidity_pools::{Pallet, Call, Storage, Event<T>} = 108,
```
Without a migration, this does not automatically translate `Connectors` prefixed storage entries to `LiquidityPools`. So we either need to write one or live with rebuilding in the LiquidityPools testnet.

NunoAlexandre replied:
I am aware but we agreed to not write migrations for `development`, right? and when we deploy the next test revision we will have the new setup with the new gateway and routers setup so we would need to set storage again anyways.

wischli replied:
Perfect! Just wanted to raise awareness as this might have been a rare exception in which we might have had to write a migration. Glad to hear we will restart fgrom scratch anyway.

NunoAlexandre replied:
Definitely, this should **never** happen in production of course. Thanks for raising it anyways!

wischli commented on runtime/integration-tests/src/liquidity_pools/pallet/development/tests/liquidity_pools.rs:
```diff
@@ -1797,7 +1802,7 @@ mod utils {
 					metadata: TrancheMetadata {
 						// NOTE: For now, we have to set these metadata fields of the first tranche
-						// to be convertible to the 32-byte size expected by the connectors
+						// to be convertible to the 32-byte size expected by the liquidityPools
```
Nit:
```suggestion
						// to be convertible to the 32-byte size expected by the liquidity pools
```

wischli reviewed (commented):
If we don't want to start from scratch on the Liquidity Pools Testnet, we unfortunately need to write a simple migration to map all `Connectors` prefixed storage items to `LiquidityPools`.

Apart from that only ridiculous nitpicks from my side and lots of love. Thanks for applying this and all the improvements! Much better than before.

NunoAlexandre commented:
@wischli thanks for the careful review and raising awareness for specific core bits. I believe to have addressed all of your comments, let me know if I missed any 🏄‍♂️

wischli reviewed (dismissed):
LGTM! 🤿

mustermeiszer commented on pallets/liquidity-pools/src/lib.rs:
```diff
@@ -348,13 +348,13 @@ pub mod pallet {
 		#[pallet::call_index(1)]
-		pub fn add_connector(origin: OriginFor<T>, connector: T::AccountId) -> DispatchResult {
+		pub fn add_instance(origin: OriginFor<T>, instance: T::AccountId) -> DispatchResult {
 			T::AdminOrigin::ensure_origin(origin)?;
-			KnownConnectors::<T>::insert(connector, ());
+			KnownInstances::<T>::insert(instance, ());
```
We do not need that here anymore, right?

NunoAlexandre replied:
Right! I also didn't expect to see this here anymore but thought that Cosmin may have had a reason to keep it. I'll address this in a follow up PR 🙏🏻

mustermeiszer commented on pallets/liquidity-pools/src/lib.rs:
```diff
@@ -266,27 +266,27 @@ pub mod pallet {
-	/// authorizing the origin of incoming messages through the `handle`
-	/// extrinsic.
+	/// The set of known instances of deployed liquidity pools across arbitrary
+	/// domains. This set is used as an allow-list when authorizing the origin
+	/// of incoming messages.
 	#[pallet::storage]
-	pub(crate) type KnownConnectors<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, ()>;
+	pub(crate) type KnownInstances<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, ()>;
```
We do not need that here anymore, right? This is all handled in the Gatway.

mustermeiszer reviewed (approved):
LGTM! 

There are some leftover artifacts in the Liquidity-Pools `lib.rs` but no blocker.

NunoAlexandre commented:
@mustermeiszer @wischli I had to merge `main` and resolve conflicts introduced by the precompiles PR which added the precompiles to the old `connectors-gateway` directory; move that here to `liquidity-pools-gateway` 👍 appreciate another round and hopefully we can get this in asap

wischli reviewed (approved):
Re-approving

mustermeiszer reviewed (approved):
Re-approve
